metadata_mapper/lambda_function.py

```diff
@@ -49,34 +53,30 @@ def parse_enrichment_url(enrichment_url):
 # {"collection_id": 26098, "mapper_type": "nuxeo", "page_filename": 2}
 # AWS Lambda entry point
 def map_page(payload, context):
-    if settings.LOCAL_RUN:
+    if settings.LOCAL_RUN and not isinstance(payload, dict):
         payload = json.loads(payload)
 
     vernacular_reader = import_vernacular_reader(payload.get('mapper_type'))
     source_vernacular = vernacular_reader(payload)
     api_resp = source_vernacular.get_api_response()
-    source_metadata_records = source_vernacular.parse(api_resp)
+    mapped_records = source_vernacular.parse(api_resp)
     collection = payload.get('collection', {})
 
-    for enrichment_url in collection.get('rikolti__pre_mapping'):
+    for enrichment_url in collection.get('rikolti__pre_mapping', []):
         enrichment_func, kwargs = parse_enrichment_url(enrichment_url)
         if not enrichment_func and settings.SKIP_UNDEFINED_ENRICHMENTS:
             continue
         logging.debug(
             f"[{collection['id']}]: running enrichment: {enrichment_func} "
             f"for page {payload['page_filename']} with kwargs: {kwargs}")
-        source_metadata_records = [
+        mapped_records = [
```

Comment:
these are the enrichments that get run prior to mapping the record.

Reply:
perhaps it is worth doing some analysis of these enrichments to see if we can just roll them in with either the parsing logic (in the case of character strippers) or the mapping logic (in the case of id assignments).

Reply:
Reverted